### DISCUSSION TITLE
Autogluon TabularPredictor in very large dataset?

### DISCUSSION THREAD

**Discussion by arminfar1** [autogluon/autogluon#2661] [Q&A] [answered]
Hi,

I am currently dealing with a very large dataset with >7MM rows and 654 features and am using TabularPredictor for a classification problem (binary label) in Python. However, I keep having memory issues and Autogluon crashes due to some memory issues. I tried to use presets = ['good_quality', 'optimize_for_deployment'] and also set the ag_args_ensemble={'num_folds_parallel': 4}, but still no luck. I tried to run my code with Sgameaker (ml.m5.24xlarge 96 CPU| 384 GiB), and I get the following err.

E0107 01:10:47.543669874   60468 chttp2_transport.cc:1103]   Received a GOAWAY with error code ENHANCE_YOUR_CALM and debug dataequal to "too_many_pings"

I am wondering if 1) I need to try a different setting or 2) Can I use Pyspark with Autoglun. 

AutoGluon Version:  0.6.1
Python Version:     3.8.16
Operating System:   Linux
Platform Machine:   x86_64
Train Data Rows:    7011252
Train Data Columns: 654

Full kwargs:
{'_feature_generator_kwargs': None,
 '_save_bag_folds': None,
 'ag_args': None,
 'ag_args_ensemble': None,
 'ag_args_fit': None,
 'auto_stack': True,
 'calibrate': 'auto',
 'excluded_model_types': None,
 'feature_generator': 'auto',
 'feature_prune_kwargs': None,
 'holdout_frac': None,
 'hyperparameter_tune_kwargs': None,
 'keep_only_best': False,
 'name_suffix': None,
 'num_bag_folds': None,
 'num_bag_sets': None,
 'num_stack_levels': None,
 'pseudo_data': None,
 'refit_full': False,
 'save_space': False,
 'set_best_to_refit_full': False,
 'unlabeled_data': None,
 'use_bag_holdout': False,
 'verbosity': 4}

**Comment by Innixma** [comment, marked as answer]
1. First try `medium_quality`. If that doesn't work
2. Try subsampling to 1/100th the rows, figure out which models you like via `predictor.leaderboard()`, then specify a custom `hyperparameters` based on those models to `.fit` using 1/10trh the data. Repeat the process, then use up-to all the data, dropping models that are bad or crash as you go.
3. Use a larger instance: m6i.32xlarge has 512 GB of memory.
4. https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-indepth.html#if-you-encounter-memory-issues
